node/build-extension.ps1

```diff
@@ -0,0 +1,51 @@
+param (
+    [Parameter(Mandatory=$true)]
+    [string]$PackageName
+)
+
+Write-Output "Building extension for $package"
+
+# Installing binary builder
+npm install node-gyp --global || $(throw "ERROR: Failed to install node-gyp")
+
+# Determine the operating system
+if ($IsMacOS) {
+    $os = "darwin"
+} elseif ($IsLinux) {
+    $os = "linux"
+    sudo apt-get install g++ make libatomic1
+} elseif ($IsWindows) {
+    $os = "win32"
+} else {
+    $os = "unknown"
+}
+
+git submodule update --init --recursive
+
+# Creating folder for future build
+New-Item -ItemType Directory -Path build | Out-Null
+
+# Renaming buiding config file
+Rename-Item -Path binding.51d -NewName binding.gyp
```

Comment:
This file will only exist in the repos which have native code (e.g. device-detection). This will to be checked before renaming. Unless this continues without logging an error?

Reply:
build-extension.ps1 called from common-ci/node/build-project.ps1 only if current package need to be builded it specified in [array]$NeedExtensions variable which passed from above i.e device-detection-node: 

`$packages = "fiftyone.devicedetection", "fiftyone.devicedetection.cloud", "fiftyone.devicedetection.onpremise", "fiftyone.devicedetection.shared";

$needExtensions = "fiftyone.devicedetection.onpremise"

./node/build-project.ps1 -RepoName $RepoName -Packages $packages -NeedExtensions $needExtensions`

Reply:
Got it. That's fine then